common/src/main/java/io/netty/util/internal/AppendableCharSequence.java

```diff
@@ -23,17 +25,11 @@
     private int pos;
 
     public AppendableCharSequence(int length) {
-        if (length < 1) {
-            throw new IllegalArgumentException("length: " + length + " (length: >= 1)");
-        }
-        chars = new char[length];
+        chars = new char[checkPositive(length, "length")];
```

Comment:
Notice that `checkPositive(...)` is functionally not the same as the `if (length < 1) { ... }` check. The new logic allows `0` while the old logic didn't.

Reply:
Yes, you're right. What is the _correct_ behaviour? The old implementation or the old Exception text?

Reply:
Got it. My mistake.